### PR TITLE
Persist opened tab between session (i.e. after application restart)

### LINKED ISSUE #503: Persist active tab between sessions

### PR DESCRIPTION
**Pull request by bmarty**
Fixes #503, after the usual cleanup

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/ui/UiStateRepository.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import im.vector.riotx.features.home.room.list.RoomListFragment
+import javax.inject.Inject
+
+
+/**
+ * This class is used to persist UI state across application restart
+ */
+class UiStateRepository @Inject constructor(private val sharedPreferences: SharedPreferences) {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/ui/UiStateRepository.kt]: Extract interface UIStateRepository and concrete implementation SharedPrefUiStateRepository

File: vector/src/main/java/im/vector/riotx/core/di/VectorComponent.kt [hunk 2]
```diff
@@ -106,6 +107,8 @@ interface VectorComponent {
 
     fun vectorFileLogger(): VectorFileLogger
 
+    fun uiStateRepository(): UiStateRepository
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/core/di/VectorComponent.kt]: No need to be there yet, as you don't use singleton scope

Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/core/di/VectorComponent.kt]: If I remove this, I get a compilation error :
```
di/ScreenComponent.java:57: error: [Dagger/MissingBinding] im.vector.riotx.features.ui.UiStateRepository cannot be provided without an @Provides-annotated method.
public abstract interface ScreenComponent {
                ^
      im.vector.riotx.features.ui.UiStateRepository is provided at
          im.vector.riotx.core.di.ScreenComponent.uiStateRepository()
  It is also requested at:
      im.vector.riotx.features.home.HomeDetailViewModel_AssistedFactory(…, uiStateRepository, …)
  The following other entry points also depend on it:
      im.vector.riotx.core.di.ScreenComponent.inject(im.vector.riotx.features.home.HomeDetailFragment)


```

Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/core/di/VectorComponent.kt]: ok, as discussed, we will rework the Dagger module in another small PR

**Pull request merged**